src/pool/srv_pool.c

```diff
@@ -3023,6 +3023,83 @@ ds_pool_svc_get_prop(uuid_t pool_uuid, d_rank_list_t *ranks,
 	return rc;
 }
 
+int
+ds_pool_reintegrate(uuid_t pool_uuid, d_rank_list_t *ranks,
+		uint32_t reint_rank, struct pool_target_id_list *reint_list)
+{
+	int				rc;
+	struct rsvc_client		client;
+	crt_endpoint_t			ep;
+	struct dss_module_info		*info = dss_get_module_info();
+	crt_rpc_t			*rpc;
+	struct pool_target_addr_list	list;
+	struct pool_add_in		*in;
+	struct pool_add_out		*out;
+	int i = 0;
+
+	rc = rsvc_client_init(&client, ranks);
+	if (rc != 0)
+		return rc;
+
+rechoose:
+
+	ep.ep_grp = NULL; /* primary group */
+	rsvc_client_choose(&client, &ep);
+
+	rc = pool_req_create(info->dmi_ctx, &ep, POOL_ADD, &rpc);
+	if (rc != 0) {
+		D_ERROR(DF_UUID": failed to create pool add rpc: "
+			""DF_RC"\n", DP_UUID(pool_uuid), DP_RC(rc));
+		D_GOTO(out_client, rc);
+	}
+
+	in = crt_req_get(rpc);
+	uuid_copy(in->pti_op.pi_uuid, pool_uuid);
+
+	rc = pool_target_addr_list_alloc(reint_list->pti_number, &list);
+	if (rc) {
+		D_ERROR(DF_UUID": pool_target_addr_list_alloc failed, rc %d.\n",
+			DP_UUID(pool_uuid), rc);
+		crt_req_decref(rpc);
+		D_GOTO(out_rpc, rc);
```

Comment:
Do you still need to do crt_req_decref here? It is taken care of at the out_rpc label.